File: app/src/main/java/com/example/meyss/javaretrofitdagger/Repository/PokemonRepo.java
package com.example.meyss.javaretrofitdagger.Repository;

import androidx.lifecycle.LiveData;
import android.content.Context;

import com.example.meyss.javaretrofitdagger.data.AppDatabase;
import com.example.meyss.javaretrofitdagger.data.Attack;
import com.example.meyss.javaretrofitdagger.data.PokAttacks;
import com.example.meyss.javaretrofitdagger.data.Pokemon;
import com.example.meyss.javaretrofitdagger.data.PokemonDAO;

import java.util.List;

import javax.inject.Inject;

import androidx.room.Room;

public class PokemonRepo {
    
    PokemonDAO dao;
    
    @Inject
    public PokemonRepo(PokemonDAO dao) {

         this.dao = dao;

    }

    public void insertAttack( List<Attack> attacks){
        dao.insertAllAttacks(attacks);


        System.out.println("pokemon Repo InsertAttacks");
    }
    public void insertPok( List<Pokemon> poks){
            dao.insertAll(poks);


        System.out.println("pokemon Repo Insert");
    }

    public Pokemon getPokemon(String name) {
        System.out.println("pokemon Repo getByName");
        return dao.findByName(name);


    }
    public List<Pokemon> getAllPoks() {
        System.out.println("pokemon Repo Get all");
        return dao.getAll();
    }

    public List<PokAttacks> getAttacks()
    {
        return dao.getPokAttack();
    }
}

File: app/src/main/java/com/example/meyss/javaretrofitdagger/Repository/PokemonRepo.kt
package com.example.meyss.javaretrofitdagger.Repository

import androidx.lifecycle.LiveData
import com.example.meyss.javaretrofitdagger.data.Attack
import com.example.meyss.javaretrofitdagger.data.PokAttacks
import com.example.meyss.javaretrofitdagger.data.Pokemon
import com.example.meyss.javaretrofitdagger.data.PokemonDAO
import javax.inject.Inject

class PokemonRepo @Inject
constructor(internal var dao: PokemonDAO) {



    val allPoks: LiveData<List<Pokemon>>
        get() {
            println("pokemon Repo Get all")
            return dao.all
        }

    val attacks: List<PokAttacks>
        get() = dao.pokAttack

    fun insertAttack(attacks: List<Attack>) {
        println("in REPOOO: " + attacks)

        dao.insertAllAttacks(attacks)


        println("pokemon Repo InsertAttacks")
    }

    fun insertPok(poks: List<Pokemon>) {
        dao.insertAll(poks)


        println("pokemon Repo Insert")
    }

    fun getPokemon(id: String): Pokemon {
        println("pokemon Repo getByName")
        return dao.findByName(id)


    }
    fun getAttacksIdPok(idpok: String):List<Attack>{

        return dao.AttacksPokId(idpok)
    }
}
